app/uk/gov/hmrc/directdebitupdateemailfrontend/controllers/SignOutController.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.directdebitupdateemailfrontend.controllers
+
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import uk.gov.hmrc.directdebitupdateemailfrontend.actions.Actions
+import uk.gov.hmrc.directdebitupdateemailfrontend.config.AppConfig
+import uk.gov.hmrc.directdebitupdateemailfrontend.views.html.TimedOut
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class SignOutController @Inject() (
+    actions:      Actions,
+    mcc:          MessagesControllerComponents,
+    appConfig:    AppConfig,
+    timedOutPage: TimedOut
+) extends FrontendController(mcc) {
+
+  def signOutFromTimeout: Action[AnyContent] = Action { implicit request =>
+    // N.B. the implicit request being passed into the page here may still have the auth
+    // token in it so take care to ensure that the sign out link is not shown by mistake
+    Ok(timedOutPage()).withNewSession
+  }
+
+  def signOut: Action[AnyContent] = actions.authenticatedJourneyAction { _ =>
+    Redirect(appConfig.BaseUrl.govUkUrl)
```

Comment:
do we not need to do `.withNewSession` here to actually log out?

Reply:
also, is this actually being used? It looks like the signing out behaviour is pointing to the url you read from config

Reply:
Oh yeah duh, will update the signout url in the template

Reply:
we probably don't need this still if we make both the timeout and the actual sign out button point to the url in config. It looks like we can't control the continue url with the auth login stub though - but we can for the real thing 

https://github.com/hmrc/bas-gateway-frontend/blob/main/conf/app.routes#L14

we should also raise a PR for app-config-prod to make sure we've got the right sign-out url there. We'd need proper test data to be able to sign in and test it in QA though 🤔

Reply:
Yeah I was going to raise some prs for app-config-<envs>. 
I'll remove the signout action from the controller now